packages/api/amplify_api/lib/src/api_plugin_impl.dart

```diff
@@ -77,17 +78,31 @@ class AmplifyAPIDart extends AmplifyAPI {
     _registerApiPluginAuthProviders();
   }
 
+  /// Register AmplifyAuthProviders that are specific to API category (API key,
+  /// OIDC or Lambda).
+  ///
   /// If an endpoint has an API key, ensure valid auth provider registered.
+  ///
+  /// Register OIDC/Lambda set to _authProviders in constructor.
   void _registerApiPluginAuthProviders() {
     _apiConfig.endpoints.forEach((key, value) {
       // Check the presence of apiKey (not auth type) because other modes might
       // have a key if not the primary auth mode.
       if (value.apiKey != null) {
         _authProviderRepo.registerAuthProvider(
-            value.authorizationType.authProviderToken,
-            AppSyncApiKeyAuthProvider());
+          value.authorizationType.authProviderToken,
+          AppSyncApiKeyAuthProvider(),
```

Comment:
On an unrelated note - why not include the API key in the auth provider, like `AppSyncApiKeyAuthProvider(value.apiKey)`? Does this mean you have to pass the API token each time you use the API key auth provider? That doesn't seem right..

Reply:
Good question. yes, the API key itself is an option passed by consumer and provider just formats request. I initially tried to implement the way you describe but it got a little complicated with multiple backends, in which case `apiName` would have to be the provided as option instead and API key provider would have to store all of them and use that apiName parameter to switch on it. It's for sure possible, I just didn't see the need for that complexity given caller will have access to the config w the key anyway.